### PR TITLE
Fix install-doc target and other gitlab failures

### PR DESCRIPTION
**Pull request by SkySkimmer**
install-doc is broken for everyone since 8995d08 (#831) so gets fixed in a separate commit.

### CODE REVIEW

**Comment by Zimmi48** [comment]
Grrrr! I told @gares several times that this commit needed to go in a separate PR otherwise it would never get reviewed. I was on the verge of asking Maxime to merge the PR without it but I didn't. It appears I should have in fact, and more importantly I shouldn't have backported the PR with it. This is a small (unpolished) feature that came in a documentation PR. What should I do for the beta? Ensure that the present PR is merged and backported? Or simply revert the faulty commit?

**Comment by Zimmi48** [comment]
For the record I've reverted 8995d08 in v8.7 (cf c2920ef). No opinion on what should become of this commit in master.

**Comment by gares** [comment]
Opps.  I think the revert is the right move to get things fixed quickly. I'm sorry I did not spot the bug.
I'm deducing that "make install" does not run "make install-doc" :-/

Said that, the fix in this PR is much more interesting (for 8.7 final). The fact that the manual you compile yourself is not the one you see online is very lame (how are you supposed to check that
it looks decent online? what is advantage of the different skin you see locally?).

Anyway the PR seems fine. The only thing worth checking that the find command works fine on OSX too.

**Comment by Zimmi48** [comment]
> The fact that the manual you compile yourself is not the one you see online is very lame (how are you supposed to check that it looks decent online? what is advantage of the different skin you see locally?).

Indeed, I agree. Although when reverting the commit I also wondered: why did you had to add CSS stylesheets to the repository? Because they were not present initially and we're stored elsewhere? If yes, how do you ensure synchronization? Wouldn't it be worth to remove these stylesheets from "elsewhere" and have the online manual use the ones from the repository? Are you sure that you need to "serve" the manual with a Python server? Can't an HTML page link to CSS stylesheets without this? These may be dumb questions but this is the kind of thing which could have been asked if this commit had gone through a separate PR. Moreover this isn't a specially relevant thing to have in v8.7 since PRs are done on master nowadays. Note that I didn't revert it in master.

**Comment by SkySkimmer** [comment]
>I'm deducing that "make install" does not run "make install-doc" :-/

It does but only with `./configure -with-doc yes`.

**Comment by SkySkimmer** [comment]
In travis we run 2 builds with coqide and doc (for 2 different ocaml versions). Maybe we should configure them with -prefix and run make install at the end.

**Comment by Zimmi48** [comment]
> The only thing worth checking that the find command works fine on OSX too.

Unfortunately, I don't think that Travis will test this so it would be great if someone with a Mac (@herbelin, @mattam82) could test this PR and confirm it correctly fixes `make install-doc`.

**Comment by gares** [comment]
@Zimmi48 the CSS are on the Coq website and for sure we need them to be in sync.
I've no access to that machine, @maximedenes has IIRC. Note that the PR about ssr manual adds a few rules in the CSS to make it look nice (IIRC hevea generates a `<listing>` node that has to be presented as a `<pre>` one, roughly).

I don't recall why a local server was needed. My guessing is that it is because some URLs are absolute, so you need the root to be `./` and not `/`.  And maybe that is the right thing to do (the absolute URL), I really don't know, but the local http server was the easy fix.

**Comment by herbelin** [comment]
This use of `find` is ok on MacOS X but not `-D` for `install`. I pushed a PR on @SkySkimmer's branch [here](https://github.com/SkySkimmer/coq/pull/2).

At the same time, I fixed and uniformized two other calls to `install -D` introduced in 8995d08 and which were not done the way it was done elsewhere in `Makefile.doc`.

**Comment by SkySkimmer** [comment]
Copy from SkySkimmer/coq#2:
I don't see how this can work considering we need to make subdirectories. Testing at https://gitlab.com/SkySkimmer/coq/pipelines/11020547
```
doc/common/styles/html/coqremote/
├── cover.html
├── footer.html
├── header.html
├── hevea.css
├── modules
│   ├── node
│   │   └── node.css
│   ├── system
│   │   ├── defaults.css
│   │   └── system.css
│   └── user
│       └── user.css
├── sites
│   └── all
│       └── themes
│           └── coq
│               ├── coqdoc.css
│               └── style.css
└── styles.hva
```
Maybe use `$(MKDIR) BLABLA/$$(dirname $$f)` instead? (not sure that's the right syntax for calling dirname)

**Comment by herbelin** [comment]
Copy from [SkySkimmer/coq#2](https://github.com/SkySkimmer/coq/pull/2#issuecomment-323610305):

@SkySkimmer: Please forget about my PR. I obviously did not understand the effect of -D and I actually did not take time to think carefully at what the target is doing. The only point I'm able to say is that -D is not available on MacOS X. Maybe a cp -r then? In any case, would it be ok for you that I cancel this PR and rely on you for finding an appropriate solution?

**Comment by SkySkimmer** [comment]
I'll do that then.

**Comment by SkySkimmer** [comment]
Gitlab is paused (https://gitlab.com/SkySkimmer/coq/pipelines/11034675) probably because they're updating but this should work.
Could someone run it on apples to make sure?

**Comment by herbelin** [comment]
@SkySkimmer:

>  Could someone run it on apples to make sure?

Seems to work for me (and the structure of the tree you showed seems respected).

**Comment by SkySkimmer** [comment]
Rebased to deal with conflict after #971.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -138,10 +142,12 @@ before_script:
...
     - DOCLIGHTDIRS="$LIB/theories/Init/ $LIB/theories/Logic/ $LIB/theories/Unicode/ $LIB/theories/Arith/"
     - DOCLIGHTVOFILES=$(find $DOCLIGHTDIRS -name '*.vo' -printf "%p ")
+    # stupid find looks in install/ ...
+    - ACCEPT_ALIEN="ACCEPT_ALIEN_VO=1 ACCEPT_ALIEN_OBJ=1"
 
-    - make doc QUICK=true COQDOC_NOBOOT=true COQTEX="$INSTALLDIR/bin/coq-tex" COQDOC="$INSTALLDIR/bin/coqdoc" VFILES="$DOCVFILES" THEORIESLIGHTVO="$DOCLIGHTVOFILES"
+    - make doc QUICK=true COQDOC_NOBOOT=true COQTEX="$INSTALLDIR/bin/coq-tex" COQDOC="$INSTALLDIR/bin/coqdoc" VFILES="$DOCVFILES" THEORIESLIGHTVO="$DOCLIGHTVOFILES" $ACCEPT_ALIEN
 
-    - make install-doc
+    - make install-doc $ACCEPT_ALIEN
```
Comment (maximedenes) [inline .gitlab-ci.yml]: Wouldn't the proper fix be to not check for alien files on install targets (in the Makefile of Coq)?

Comment (maximedenes) [reply .gitlab-ci.yml]: @SkySkimmer WDYT?

Comment (SkySkimmer) [reply .gitlab-ci.yml]: Adding the folder gitlab uses for installing things in to `FIND_SKIP_DIRS` is also a possibility.

@letouzey By the way why are there separate ACCEPT_ALIEN_FOO variables?

Comment (SkySkimmer) [reply .gitlab-ci.yml]: I don't feel like messing with Makefile flow control so I just added it to `FIND_SKIP_DIRS`.

**Comment by SkySkimmer** [comment]
Running at https://gitlab.com/SkySkimmer/coq/pipelines/11696355

**Comment by SkySkimmer** [comment]
I forgot one install -> _install_ci, new run at https://gitlab.com/SkySkimmer/coq/pipelines/11698419
- sf is failing because I didn't rebase on master so it's fetching the wrong URL
- the test suite is failing in coq-makefile/findlib-package/run.sh
```
make[1]: Entering directory '/builds/SkySkimmer/coq/test-suite/coq-makefile/findlib-package'
make[2]: Entering directory '/builds/SkySkimmer/coq/test-suite/coq-makefile/findlib-package'
CAMLC -c src/test_aux.ml
CAMLC -pp -c src/test.ml4
CAMLC -pack -o src/test_plugin.cmo
CAMLC -a -o src/test_plugin.cma
COQC theories/test.v
/builds/SkySkimmer/coq/_install_ci/bin/coqc: "execvp" failed on "/builds/SkySkimmer/coq/_install_ci/bin/coqtop.byte": No such file or directory
Makefile:637: recipe for target 'theories/test.vo' failed
```
`coqtop.byte` doesn't get installed, but I don't understand why this failure only happened in the `test-suite` job and not in the 32bit test suite or the ocaml 4.05 test suite jobs. The previous run had the test suite fail in the same way but only in the 32bit job.
Does anyone familiar with coq_makefile have any clue? @gares?

**Comment by Zimmi48** [comment]
I'm going to backport only commit 741943da6e42937f6d50db9c920a40073160eebc since the other one fixes an issue that was now only in master.

**Pull request merged**